docs/intro/intro_samplers.rst

```diff
@@ -0,0 +1,96 @@
+.. _intro_samplers:
+
+=======================
+Samplers and Composites
+=======================
+
+dimod includes reference :term:`sampler`\ s and :term:`composite`\ s for processing
+binary quadratic models (and higher order models) and refining sampling, and for
+testing your code during development.
+
+For an introduction to samplers and composites, see
+:std:doc:`Samplers and Composites <oceandocs:concepts/samplers>`.
+
+For descriptions of all supported samplers, see :ref:`samplers_index`.
+
+Using Samplers
+==============
+
+To find solutions to the small four-node
+`maximum cut <https://en.wikipedia.org/wiki/Maximum_cut>`_
+BQM generated in :ref:`intro_bqm`, shown again in the figure below, you can use
+one of dimod's reference samplers: its :class:`~dimod.reference.samplers.ExactSolver`
+test sampler, for example, calculates the energy of all possible samples.
+
+.. figure:: ../_images/four_node_star_graph.png
+    :align: center
+    :scale: 40 %
+    :name: four_node_star_graph2
+    :alt: Four-node star graph
+
+    Star graph with four nodes.
+
+>>> qubo = dict({(0, 0): -3, (1, 1): -1, (0, 1): 2, (2, 2): -1,
+...              (0, 2): 2, (3, 3): -1, (0, 3): 2})
+>>> dict_bqm = dimod.BQM.from_qubo(qubo)
+>>> sampler_exact = dimod.ExactSolver()
+>>> sampleset = sampler_exact.sample(dict_bqm)
+>>> print(sampleset)
+    0  1  2  3 energy num_oc.
+1   1  0  0  0   -3.0       1
+11  0  1  1  1   -3.0       1
+2   1  1  0  0   -2.0       1
+...
+10  1  1  1  1    0.0       1
+['BINARY', 16 rows, 16 samples, 4 variables]
+
+Using Composed Samplers
+=======================
+
+This example uses a composed sampler on the
+:std:doc:`Boolean NOT Gate <oceandocs:examples/not>`
+example detailed in the :std:doc:`Getting Started <oceandocs:getting_started>`
+documentation.
+
+The :class:`~dimod.reference.composites.fixedvariable.FixedVariableComposite`
+composite sets the value and removes specified variables from the BQM before
+sending it to the sampler. 
+
+>>> from dimod import FixedVariableComposite, ExactSolver
+>>> Q = {('x', 'x'): -1, ('x', 'z'): 2, ('z', 'x'): 0, ('z', 'z'): -1}
```

Comment:
I have personally stayed away from that format in examples etc, worried that it might be confusing or that we might see things like `{'var1var2': 1}` which obviously won't work. Happy to defer to the doc expert though

Reply:
I agree, that's why I said I'm not sure about the pedagogical aspect. :smile:

ZoP#3,5,7,9 might be in support of the simpler variant, but I'm sure one could find the same number of cons.

Reply:
I like the format for concise doctrings examples in non-basic function etc but for a RTD intro the audience likely includes users without Python expertise for whom this is another step to parse